### ISSUE TITLE
RBENV_VERSION ignored when shelling out from ruby

### ISSUE THREAD

**Issue by bronson** [rbenv/rbenv#121]
This shows that RBENV_VERSION works when executed from Perl but not from Ruby.  In this example, global version is set to 1.9.2p290.

```
$ ruby --version
ruby 1.9.2p290 (2011-07-09 revision 32553) [x86_64-linux]
$ perl -e 'system("RBENV_VERSION=system ruby -e \"puts RUBY_VERSION\"")'
1.8.7
$ ruby -e 'system("RBENV_VERSION=system ruby -e \"puts RUBY_VERSION\"")'
1.9.2
```

RBENV_VERSION should always work, shouldn't it?

**Comment by mislav** [comment]
The fix of this was reverted in 95a039aaaa3855ea2df4855ad38c06faaba01f9a so I'm reopening this.

**Comment by mislav** [comment]
Closing this because it's now documented as one of the pitfalls of the design of rbenv: https://github.com/rbenv/rbenv/wiki/Comparison-of-version-managers

I've had more than a decade to mull this over and there were attempts to fix, but that broke other things, so I think it's best to accept that with rbenv, once a Ruby version is activated in a specific process, the only way to switch to a different Ruby version within that process is to restore the original PATH.
